lib/ssl/src/ssl_handshake.erl

```diff
@@ -2648,6 +2724,31 @@ decode_extensions(<<?UINT16(?PRE_SHARED_KEY_EXT), ?UINT16(Len),
                                #pre_shared_key_server_hello{
                                   selected_identity = Identity}});
 
+%% RFC6066, if a server returns a "CertificateStatus" message, then
+%% the server MUST have included an extension of type "status_request"
+%% with empty "extension_data" in the extended server hello.
+decode_extensions(<<?UINT16(?STATUS_REQUEST), ?UINT16(Len),
+                    _ExtensionData:Len/binary, Rest/binary>>, Version,
+                    MessageType = server_hello, Acc)
+  when Len =:= 0 ->
+    decode_extensions(Rest, Version, MessageType,
+                      Acc#{ocsp_stapling => undefined});
+%% RFC8446 4.4.2.1, In TLS1.3, the body of the "status_request" extension
+%% from the server MUST be a CertificateStatus structure as defined in
+%% RFC6066.
+decode_extensions(<<?UINT16(?STATUS_REQUEST), ?UINT16(Len),
+                    CertStatus:Len/binary, Rest/binary>>, Version,
+                    MessageType, Acc) ->
+    case CertStatus of
+        <<?BYTE(?CERTIFICATE_STATUS_TYPE_OCSP),
+          ?UINT24(OCSPLen),
+          ASN1OCSPResponse:OCSPLen/binary>> ->
+            decode_extensions(Rest, Version, MessageType,
+                      Acc#{ocsp_stapling => ASN1OCSPResponse});
+        _Other ->
+            decode_extensions(Rest, Version, MessageType, Acc)
```

Comment:
At this point we have recognized a `status_request` extension but it contains invalid data. I think we should probably abort the handshake instead of dropping the extension @IngelaAndin ?
```erlang
throw(?ALERT_REC(?FATAL, ?HANDSHAKE_FAILURE, {malformed_status_request, Other})
```
But we should test if this works or not. This would be a better way to handle protocol errors.